record/record.go

```diff
@@ -209,6 +222,7 @@ func (r *Reader) Next() (io.Reader, error) {
 // Recover also marks the current reader, the one most recently returned by
 // Next, as stale.
 func (r *Reader) Recover() {
+	r.inRecovery = true
```

Comment:
Should Recover be a no-op if r.err == nil ?  If you call it without an error, you'll end up discarding the current block and any subsequent blocks.  I had this in the last PR, but I ended up taking it out.  What do you think?